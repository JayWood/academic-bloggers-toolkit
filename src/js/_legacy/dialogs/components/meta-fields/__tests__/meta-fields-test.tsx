import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';

import ManualDataStore from '_legacy/stores/data/manual-data-store';
import MetaFields from '..';

const setup = (citationType: CSL.ItemType = 'webpage') => {
    const store = new ManualDataStore(citationType);
    const component = shallow(<MetaFields meta={store} />);
    return {
        component,
        instance: component.instance() as MetaFields,
    };
};

describe('<MetaFields />', () => {
    it('should match snapshots', () => {
        const { component } = setup();
        expect(
            toJSON(component, {
                noKey: true,
                map: json => ({
                    ...json,
                    props: { ...json.props, meta: '__SKIPPED__' },
                }),
            }),
        ).toMatchSnapshot();
    });
    it('should update fields', () => {
        const { instance } = setup();
        expect(instance.props.meta.fields.title).toBe('');
        instance.updateField({
            currentTarget: { dataset: { field: 'title' }, value: 'testing' },
        } as any);
        expect(instance.props.meta.fields.title).toBe('testing');
    });
    it('should throw errors when field ID is not set properly', () => {
        const { instance } = setup();
        expect(() =>
            instance.updateField({ currentTarget: { dataset: {} } } as any),
        ).toThrowError(ReferenceError);
    });
});
