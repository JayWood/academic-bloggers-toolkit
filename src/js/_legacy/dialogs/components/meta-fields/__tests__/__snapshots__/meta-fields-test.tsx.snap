// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaFields /> should match snapshots 1`] = `
<React.Fragment
  meta="__SKIPPED__"
>
  <h2
    className="heading"
    meta="__SKIPPED__"
  >
    Web Page
  </h2>
  <div
    className="table"
    meta="__SKIPPED__"
  >
    <Field
      field={
        Object {
          "label": "Content Title",
          "required": true,
          "value": "title",
        }
      }
      meta="__SKIPPED__"
      onChange={[Function]}
    />
    <Field
      field={
        Object {
          "label": "Website Title",
          "required": true,
          "value": "container-title",
        }
      }
      meta="__SKIPPED__"
      onChange={[Function]}
    />
    <Field
      field={
        Object {
          "label": "URL",
          "required": true,
          "value": "URL",
        }
      }
      meta="__SKIPPED__"
      onChange={[Function]}
    />
    <Field
      field={
        Object {
          "label": "Date",
          "pattern": "[0-2][0-9]{3}(?:(\\\\/(?:0[1-9]|1[0-2]))(\\\\/(?:[0-2][0-9]|3[0-1]))?)?",
          "title": "YYYY/MM/DD or YYYY/MM or YYYY",
          "value": "issued",
        }
      }
      meta="__SKIPPED__"
      onChange={[Function]}
    />
    <Field
      field={
        Object {
          "label": "Date Accessed",
          "pattern": "[0-2][0-9]{3}(?:(\\\\/(?:0[1-9]|1[0-2]))(\\\\/(?:[0-2][0-9]|3[0-1]))?)?",
          "title": "YYYY/MM/DD or YYYY/MM or YYYY",
          "value": "accessed",
        }
      }
      meta="__SKIPPED__"
      onChange={[Function]}
    />
  </div>
</React.Fragment>
`;
